### DISCUSSION TITLE
Force to use lower case URL?

### DISCUSSION THREAD

**Discussion by orchardc** [OrchardCMS/OrchardCore#6951] [General]
Hi,

I have tried to use the following code in startup.cs , it works fine in my custom model but it break in the Admin page (For example, when you try to create a new page). It looks like Orchard Admin page is URL senstive.

So the question is how to force the whole application to use lower case URL?

`services.Configure<RouteOptions>(options => options.LowercaseUrls = true)`

**Comment by Skrypt** [comment]
There is a configurable url prefix for the admin.

OrchardCore.Admin.Abstractions --> AdminOptions class.

By default the _adminUrlPrefix is set to "Admin" so on Linux it would respond to /Admin only.
This route can be configured as documented here : 

https://docs.orchardcore.net/en/dev/docs/reference/modules/Admin/#custom-admin-prefix

Also some more details explained in this issue : https://github.com/OrchardCMS/OrchardCore/issues/6889


I believe that if you would want to enforce lowercase url's throughout the app you would need to add this 
`services.Configure<RouteOptions>(options => options.LowercaseUrls = true)`
in each Startup.cs files of each OC module and make it an actual global site setting option. 
This is worth opening an issue so that it be at least evaluated.

**Reply by orchardc** [reply]
Thanks @Skrypt , but the problem is still there after I put the code into each startup.cs file.

![error](https://user-images.githubusercontent.com/16193516/91105571-42355480-e668-11ea-94dd-10913b86e3ff.JPG)

**Reply by deanmarcussen** [reply]
Yeah I don't think that will work, because the `ContentType` is a case sensitive parameter, i.e. `Page` is considered different to `page`.

In this case it's less about the route, and more about internally `ContentType` being case sensitive.

**Reply by Skrypt** [reply]
Maybe related to the Content Type names but then where these routes are defined? If the module/service registering them defines that routes should be lowercased then it should lowercase them even if the technical name of those content types contains upper case letters.

I think the issue is that you are missing one place where a service is registered. Maybe in the DisplayManager project.
